src/components/refactor/dialogs/connectivity/connectivity-form.js

```diff
@@ -0,0 +1,272 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import Grid from '@mui/material/Grid';
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+    fetchBusbarSectionsForVoltageLevel,
+    fetchBusesForVoltageLevel,
+    fetchVoltageLevels,
+} from '../../../../utils/rest-api';
+import { useSelector } from 'react-redux';
+import ExploreOutlinedIcon from '@mui/icons-material/ExploreOutlined';
+import ExploreOffOutlinedIcon from '@mui/icons-material/ExploreOffOutlined';
+
+import PositionDiagramPane from '../../../diagrams/singleLineDiagram/position-diagram-pane';
+import { IconButton, Tooltip } from '@mui/material';
+import { CONNECTION_DIRECTIONS } from '../../../network/constants';
+import { useIntl } from 'react-intl';
+import { isNodeBuilt } from '../../../graph/util/model-functions';
+import { useFormContext } from 'react-hook-form';
+import {
+    BUS_OR_BUSBAR_SECTION,
+    CONNECTION_DIRECTION,
+    CONNECTION_NAME,
+    CONNECTION_POSITION,
+    CONNECTIVITY,
+    getConnectivityBusBarSectionData,
+    getConnectivityVoltageLevelData,
+    VOLTAGE_LEVEL,
+} from './connectivity-form-utils';
+import TextInput from '../../rhf-inputs/text-input';
+import SelectInput from '../../rhf-inputs/select-input';
+import IntegerInput from '../../rhf-inputs/integer-input';
+import AutocompleteInput from '../../rhf-inputs/autocomplete-input';
+
+/**
+ * Hook to handle a 'connectivity value' (voltage level, bus or bus bar section)
+ * @param id optional id that has to be defined if the hook is used more than once in a form
+ * @param direction direction of placement. Either 'row' or 'column', 'row' by default.
+ * @param withPosition
+ * @param withDirectionsInfos
+ * @returns {[{voltageLevel: null, busOrBusbarSection: null},unknown]}
+ */
+export const ConnectivityForm = ({
+    id,
+    direction = 'row',
+    withDirectionsInfos = true,
+    withPosition = false,
+}) => {
+    const studyUuid = useSelector((state) => state.studyUuid);
+    const currentNode = useSelector((state) => state.currentTreeNode);
```

Comment:
`studyUuid currentNode` it would be better to pass them in props because they don't change when the modal is rendered